search-service/src/main/kotlin/com/egm/stellio/search/authorization/permission/web/PermissionHandler.kt

```diff
@@ -326,16 +318,26 @@ class PermissionHandler(
                     Unit.right()
             }
 
-    private suspend fun checkIsAdmin(permission: Permission): Either<APIException, Unit> =
-        permissionService.checkHasPermissionOnEntity(permission.target.id, Action.ADMIN)
-            .flatMap {
-                if (!it)
-                    AccessDeniedException(
-                        unauthorizedCreateMessage(permission.target.id)
-                    ).left()
-                else
-                    Unit.right()
+    private suspend fun checkCanModify(
```

Comment:
Having a `checkCanCreate` and `checkCanDelete` will make things a lot more clear (and will actually duplicate only two lines of code)